### ISSUE TITLE
Input field accepts multiple characters in single field for inputType number.

### ISSUE THREAD

**Issue by l3ikrant** [devfolioco/react-otp-input#393]
**Steps to reproduce the issue**:
- Navigate to live demo url (https://devfolioco.github.io/react-otp-input/)
- Change inputType to number
- Start typing  characters

 **Screenshot**:
![2023-04-10_19-02](https://user-images.githubusercontent.com/7446485/230908776-c41c9539-29d0-4acf-affe-a2685bb73395.png)

**Comment by prateek3255** [comment]
I am not able to reproduce this @l3ikrant can you add a video demo, on how you did it?

**Comment by l3ikrant** [comment]
***Video Link***
https://www.awesomescreenshot.com/video/16380479?key=380150336acb2a5f625ce75e7f034e3f

**Comment by l3ikrant** [comment]
Right! I did testing with other browsers like Chrome and brave, and it is working fine. The issue persists only with Firefox Web Browser.

**Comment by prateek3255** [comment]
Okay, would you like to work on a fix for this issue?

**Comment by bVatsaL** [comment]
@l3ikrant @prateek3255  I have the same issue and it can be resolved by adding `onKeyDown` function to our own input with the validation below.
```
<OtpInput
    value={otp}
    onChange={setOtp}
    numInputs={4}
    inputType="number"
    renderInput={(props) => (
      <input
        {...props}
        onKeyDown={handleKeyDown}
      />
    )}
  />
```
```
const handleKeyDown: KeyboardEventHandler<HTMLInputElement> = (e) => {
    if (isNaN(+e.key) && !VALID_KEYS.includes(e.key)) {
      e.preventDefault();
    }
  };
```
```
const VALID_KEYS = [
  'Backspace',
  'Enter',
  'ArrowDown',
  'ArrowUp',
  'ArrowLeft',
  'ArrowRight',
  'Shift',
  'Tab',
];
```

**Comment by prateek3255** [comment]
@l3ikrant @bVatsaL This issue has been fixed with v3.0.2. On a sidenote @bVatsaL you should not override props like `onKeyDown` this will lead to unexpected behaviours with the OTP input, as a caution I have also added a [warning section in the README](https://github.com/devfolioco/react-otp-input#%EF%B8%8F-warning), specifying which props you should not override.
